core/include/ikos/core/domain/memory/value.hpp

```diff
@@ -1162,6 +1164,60 @@ class ValueDomain final
     return updated_cells;
   }
 
+
+  /// \brief Detect Read from Uninitialized Memory
+  bool is_read_uninitialized(MemoryLocationRef base,
+                            const MachineInt& offset,
+                            const MachineInt& size) {
+    CellSetT cells = this->_cells.get(base);
+    if (cells.is_empty()) {
+      return true; // If no cells, certainly uninitialized.
+    }
+    if (!offset.fits<long>()) {
+      // Give up if we cannot manipulate the offset natively.
+      return false;
+    }
+    if (!size.fits<long>()) {
+      // Give up if we cannot manipulate the size natively.
+      return false;
+    }
```

Comment:
As you suggested, my next push will use z numbers instead of bit sets. My revised algorithm only needs bit sets as big as the size being read. The z number infrastructure does not support shifting by more than the maximum unsigned 32 bit number, so there is that limitation, but I don't think reads of more then 4GB are very likely.